File: CS-Examples/15_MarkerDesigner/DetectIsBlank.cs
using System;
using System.Data;
using System.Windows.Forms;
using Spire.Xls;

namespace DetectIsBlank
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            // Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\MarkerDesigner2.xlsx");

            // Create a DataSet
            DataSet ds = new DataSet();

            // Fill the DataSet from an XML file
            ds.ReadXml(@"..\..\..\..\..\..\Data\Data.xml");

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Fill a DataTable using the "data" parameter in Marker Designer
            workbook.MarkerDesigner.AddDataTable("data", ds.Tables["data"]);
            workbook.MarkerDesigner.Apply();

            // Calculate all formulas in the workbook
            workbook.CalculateAllValue();

            // Save the modified workbook to a file named "DetectIsBlank.xlsx" using Excel 2013 format
            string output = "DetectIsBlank.xlsx";
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: CS-Examples/15_MarkerDesigner/DetectIsBlank.vb
Imports Spire.Xls

Namespace DetectIsBlank

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Load an existing workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\MarkerDesigner2.xlsx")

            ' Create a new DataSet object
            Dim ds As New DataSet()

            ' Read XML data from the specified file path into the DataSet
            ds.ReadXml("..\..\..\..\..\..\Data\Data.xml")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Add the DataTable from the DataSet with the name "data" to the MarkerDesigner
            workbook.MarkerDesigner.AddDataTable("data", ds.Tables("data"))

            ' Apply the marker designer to replace the markers with data from the DataTable
            workbook.MarkerDesigner.Apply()

            ' Calculate all the formulas in the workbook
            workbook.CalculateAllValue()

            ' Define the output file name as "DetectIsBlank.xlsx"
            Dim output As String = "DetectIsBlank.xlsx"

            ' Save the modified workbook to the specified file path using Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
